src/main/java/io/openepcis/convert/Conversion.java

```diff
@@ -0,0 +1,90 @@
+package io.openepcis.convert;
+
+import io.openepcis.constants.EPCISFormat;
+import io.openepcis.constants.EPCISVersion;
+import lombok.Getter;
+
+@Getter
+public class Conversion {
+
+    private EPCISFormat fromMediaType;
+    private EPCISVersion fromVersion;
+    private EPCISFormat toMediaType;
+    private EPCISVersion toVersion;
+    private boolean generateGS1CompliantDocument;
+
+    private Conversion() {
+    }
+
+    public static FromMediaTypeStage newBuilder() {
+        return new Stages();
+    }
+
+    public interface FromMediaTypeStage {
+        ToVersionStage fromMediaType(EPCISFormat fromMediaType);
+    }
+
+    public interface ToVersionStage {
+        ToVersionStage fromVersion(EPCISVersion fromVersion);
+        GS1ComplianceStage toVersion(EPCISVersion toVersion);
+        ToVersionStage toMediaType(EPCISFormat toMediaType);
+    }
+
+    public interface GS1ComplianceStage {
+        BuildStage generateGS1CompliantDocument(boolean generateGS1CompliantDocument);
+    }
+    public interface BuildStage {
+        Conversion build();
+    }
+
+    private static class Stages implements FromMediaTypeStage, ToVersionStage, BuildStage, GS1ComplianceStage {
+
+        private EPCISFormat fromMediaType;
+        private EPCISVersion fromVersion;
+        private EPCISFormat toMediaType;
+        private EPCISVersion toVersion;
+        private boolean generateGS1CompliantDocument;
+
+        @Override
+        public ToVersionStage fromMediaType(EPCISFormat fromMediaType) {
+            this.fromMediaType = fromMediaType;
+            return this;
+        }
+
+        @Override
+        public ToVersionStage fromVersion(EPCISVersion fromVersion) {
+            this.fromVersion = fromVersion;
+            return this;
+        }
+
+        @Override
+        public ToVersionStage toMediaType(EPCISFormat toMediaType) {
+            this.toMediaType = toMediaType;
+            return this;
+        }
+
+        @Override
+        public GS1ComplianceStage toVersion(EPCISVersion toVersion) {
+            this.toVersion = toVersion;
+            return this;
+        }
+
+        @Override
+        public BuildStage generateGS1CompliantDocument(boolean generateGS1CompliantDocument) {
+            this.generateGS1CompliantDocument = generateGS1CompliantDocument;
+            return this;
+        }
+
+        @Override
+        public Conversion build() {
+            Conversion conversion = new Conversion();
```

Comment:
the reason for having stages is to have all elements in definite state.
checking for null for assigning alternative should not be required